### PR TITLE
Ports psoc6 spi

### PR DESCRIPTION
**Pull request by IFX-Anusha**
By creating this pull request you agree to the terms in CONTRIBUTING.md.
https://github.com/Infineon/.github/blob/master/CONTRIBUTING.md
--- DO NOT DELETE ANYTHING ABOVE THIS LINE ---

CONTRIBUTING.md also tells you what to expect in the PR process.

Description
SPI Module implementation, tests & Documentation.
NB: Only Master mode is implemented

### CODE REVIEW

**Review by jaenrig-ifx — APPROVED** [review]
Looks good! :)

File: ports/psoc6/mphalport.c [hunk 1]
```diff
@@ -121,6 +121,22 @@ uint8_t mp_hal_pin_name(mp_hal_pin_obj_t pin) {
     return pin;
 }
 
+void mp_hal_pin_write(mp_hal_pin_obj_t pin, uint8_t polarity) {
+    if (polarity == 1) {
+        gpio_set_value(pin);
+    } else {
+        gpio_get_value(pin);
```
Comment (bargfred) [inline ports/psoc6/mphalport.c:128]: Why is this required if the return value is not used anyway ?

File: docs/psoc6/quickref.rst [hunk 2]
```diff
@@ -347,3 +347,83 @@ The constructor can be called by passing the required arguments. All initializat
...
+    from machine import SPI
+    spi = SPI(0) # Default assignment: id=0, sck=P9_2  ,MOSI=P9_0, MISO=P9_1, baudrate=1000000, Polarity=0, Phase=0, bits=8, firstbit=SPI.MSB
+    spi.init()
+
+Management of a CS signal should happen in user code (via machine.Pin class).
+
+::
+    from machine import Pin
+    cs = Pin('P9_3', mode=Pin.OUT, value=1)      # Create chip-select on pin P9_3
+    cs(0)                                        # select the peripheral
```
Comment (bargfred) [inline docs/psoc6/quickref.rst:416]: I do not understand this command : cs(0) on a pin object. What does this do ?

**Review by bargfred — APPROVED** [review]

File: docs/psoc6/quickref.rst (hunk 2 above)
Comment (IFX-Anusha) [reply docs/psoc6/quickref.rst:416]: This is the chip select pin which is an active low signal. So by cs(0) it will select the pin corresponding pin & makes it value=0.

File: ports/psoc6/mphalport.c (hunk 1 above)
Comment (IFX-Anusha) [reply ports/psoc6/mphalport.c]: Thanks! Yes It's fixed . It is gpio_clear_value ()

File: docs/psoc6/quickref.rst (hunk 2 above)
Comment (bargfred) [reply docs/psoc6/quickref.rst:416]: Why do you not set it to 0 in the line above ?

Comment (IFX-Anusha) [reply docs/psoc6/quickref.rst:416]: https://docs.micropython.org/en/latest/library/machine.SPI.html 
Here in the common example given by micropython, it's like that. So in order to follow the same I did that.

Comment (jaenrig-ifx) [reply docs/psoc6/quickref.rst:416]: It makes sense to initialize it high (not selected), and the select the device later pulling the line low no?
It makes sense how is implemented by @IFX-Anusha.
But of course, you could directly initialize it low, if you directly intent to select that device. But this is more illustrative I feel.
